src/test/java/org/sonarlint/intellij/AbstractSonarLintLightTests.java

```diff
@@ -135,12 +137,27 @@ public void loadToolWindow() {
 
   protected void connectProjectTo(String hostUrl, String connectionName, String projectKey) {
     ServerConnection connection = ServerConnection.newBuilder().setHostUrl(hostUrl).setName(connectionName).build();
-    globalSettings.addServerConnection(connection);
-    projectSettings.bindTo(connection, projectKey);
+    getGlobalSettings().addServerConnection(connection);
+    getProjectSettings().bindTo(connection, projectKey);
   }
 
   protected void connectProjectTo(ServerConnection connection, String projectKey) {
-    globalSettings.addServerConnection(connection);
-    projectSettings.bindTo(connection, projectKey);
+    getGlobalSettings().addServerConnection(connection);
+    getProjectSettings().bindTo(connection, projectKey);
+  }
+
+  protected void setProjectLevelExclusions(List<String> exclusions) {
+    SonarLintProjectSettings projectSettings = getProjectSettings();
+    projectSettings.setFileExclusions(exclusions);
+    ProjectConfigurationListener projectListener = getProject().getMessageBus().syncPublisher(ProjectConfigurationListener.TOPIC);
+    projectListener.changed(projectSettings);
+  }
+
+  protected void setGlobalLevelExclusions(List<String> exclusions) {
+    SonarLintGlobalSettings globalSettings = getGlobalSettings();
+    globalSettings.setFileExclusions(exclusions);
+    GlobalConfigurationListener globalConfigurationListener = ApplicationManager.getApplication()
+      .getMessageBus().syncPublisher(GlobalConfigurationListener.TOPIC);
+    globalConfigurationListener.applied(globalSettings);
```

Comment:
I find it strange that we have to take care of this in the test code, we probably do this in production code at some point after the settings are changed ?

Reply:
This is what we do in the production code. We could extract that in a helper, but for now I think it's ok to keep it like this.